src/main/java/io/openliberty/tools/common/arquillian/util/HttpPortUtil.java

```diff
@@ -46,16 +47,21 @@ public class HttpPortUtil {
     public static final int DEFAULT_PORT = 9080;
     private static final XPath XPATH = XPathFactory.newInstance().newXPath();
 
-    private static final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-    private static boolean factoryInitialized = false;
-
-    public static void initDocumentBuilderFactory() throws ParserConfigurationException {
-        if (!factoryInitialized) {
-            factory.setNamespaceAware(true);
-            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false); 
-            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);    
-        }
-   }
+    public static DocumentBuilderFactory getDocumentBuilderFactory() throws ParserConfigurationException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+
+        factory.setNamespaceAware(true);
+        factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+        factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+        factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+        factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+        factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        factory.setXIncludeAware(false);
+        factory.setNamespaceAware(true);
+        factory.setExpandEntityReferences(false);
+        return factory;
```

Comment:
Are we not allowed to have a singleton instance of the factory? We were only doing the initialization once before. This can get called many times from maven/gradle plugins. This new code is creating a new one every single time.